extensions/hibernate-orm/deployment/src/test/java/io/quarkus/hibernate/orm/dev/HibernateOrmDevJsonRpcServiceTest.java

```diff
@@ -0,0 +1,127 @@
+
+import io.quarkus.devui.tests.DevUIJsonRPCTest;
+import io.quarkus.hibernate.orm.TestTags;
+import io.quarkus.test.QuarkusDevModeTest;
+
+@Tag(TestTags.DEVMODE)
+public class HibernateOrmDevJsonRpcServiceTest extends DevUIJsonRPCTest {
+    @RegisterExtension
+    final static QuarkusDevModeTest TEST = new QuarkusDevModeTest()
+            .withApplicationRoot((jar) -> jar.addClasses(MyEntityWithSuccessfulDDLGeneration.class)
+                    .addAsResource("application.properties")
+                    .addAsResource("import-custom-table-name.sql", "import.sql"));
+
+    public HibernateOrmDevJsonRpcServiceTest() {
+        super("io.quarkus.quarkus-hibernate-orm");
+    }
+
+    @Test
+    public void testGetInfo() throws Exception {
+        JsonNode info = super.executeJsonRPCMethod("getInfo");
+
+        assertThat(info).isNotNull();
+        assertThat(info.has("persistenceUnits")).isTrue();
+
+        JsonNode persistenceUnits = info.get("persistenceUnits");
+        assertThat(persistenceUnits.isArray()).isTrue();
+        assertThat(persistenceUnits.size()).isEqualTo(1);
+
+        JsonNode unit = persistenceUnits.get(0);
+        assertThat(unit.get("name").textValue()).isEqualTo("<default>");
+
+        assertThat(unit.get("managedEntities").size()).isEqualTo(1);
+        JsonNode myEntity = unit.get("managedEntities").get(0);
+        assertThat(myEntity.get("name").textValue()).isEqualTo(MyEntityWithSuccessfulDDLGeneration.NAME);
+        assertThat(myEntity.get("className").textValue()).isEqualTo(MyEntityWithSuccessfulDDLGeneration.class.getName());
+        assertThat(myEntity.get("tableName").textValue()).isEqualTo(MyEntityWithSuccessfulDDLGeneration.TABLE_NAME);
+
+        assertThat(unit.get("namedQueries").size()).isEqualTo(1);
+        JsonNode namedQuery = unit.get("namedQueries").get(0);
+        assertThat(namedQuery.get("name").textValue()).isEqualTo("MyEntity.findAll");
+        assertThat(namedQuery.get("query").textValue()).isEqualTo("SELECT e FROM MyEntity e ORDER BY e.name");
+    }
+
+    @Test
+    public void testGetNumberOfPersistenceUnits() throws Exception {
+        JsonNode units = super.executeJsonRPCMethod("getNumberOfPersistenceUnits");
+
+        assertThat(units).isNotNull();
+        assertThat(units.isNumber()).isTrue();
+        assertThat(units.intValue()).isEqualTo(1);
+    }
+
+    @Test
+    public void testGetNumberOfEntityTypes() throws Exception {
+        JsonNode types = super.executeJsonRPCMethod("getNumberOfEntityTypes");
+
+        assertThat(types).isNotNull();
+        assertThat(types.isNumber()).isTrue();
+        assertThat(types.intValue()).isEqualTo(1);
+    }
+
+    @Test
+    public void testGetNumberOfNamedQueries() throws Exception {
+        JsonNode queries = super.executeJsonRPCMethod("getNumberOfNamedQueries");
+
+        assertThat(queries).isNotNull();
+        assertThat(queries.isNumber()).isTrue();
+        assertThat(queries.intValue()).isEqualTo(2);
+    }
```

Comment:
FWIW this is already tested in integration tests, e.g. https://github.com/quarkusio/quarkus/blob/4834e06c752639ad4a547622d5541775a6d3e8b8/integration-tests/devmode/src/test/java/io/quarkus/test/devui/AbstractDevUIHibernateOrmTest.java#L20-L19

@phillip-kruger should we move all these tests to the `deployment` module instead?

Reply:
I've completely missed the existing test. I've moved the tests for the `executeHQL` method there for now.

+1 from me to move the test, maybe not under `deployment` but under `hibernate-orm/runtime`, where `HibernateOrmDevJsonRpcService` is. Though I'm fine with keeping it under ITs as well.